packages/dev/core/src/Audio/v2/abstractAudioNode.ts

```diff
@@ -0,0 +1,186 @@
+/* eslint-disable babylonjs/available */
+/* eslint-disable jsdoc/require-jsdoc */
+
+import type { AbstractAudioEngine } from "./abstractAudioEngine";
+import { AbstractAudioNodeParent } from "./abstractAudioNodeParent";
+import type { Nullable } from "../../types";
+
+export enum AudioNodeType {
+    /**
+     * Input nodes receive audio data from an upstream node.
+     */
+    Input = 1,
+
+    /**
+     * Output nodes send audio data to a downstream node.
+     */
+    Output = 2,
+
+    /**
+     * Input/Output nodes receive audio data from an upstream node and send audio data to a downstream node.
+     */
+    InputOutput = 3,
+}
+
+export abstract class AbstractAudioNode extends AbstractAudioNodeParent {
+    // If parent is null, node is owned by audio engine.
+    private _parent: Nullable<AbstractAudioNodeParent> = null;
+
+    public readonly engine: AbstractAudioEngine;
+
+    /**
+     * The connected downstream audio nodes.
+     *
+     * Undefined for input nodes.
+     */
+    protected readonly _connectedDownstreamNodes?: Set<AbstractAudioNode> | undefined;
```

Comment:
This is redundant because of the `?`.

```suggestion
    protected readonly _connectedDownstreamNodes?: Set<AbstractAudioNode>;
```

Reply:
Ya, it seems strange to me, too, but we discussed it earlier in https://github.com/BabylonJS/Babylon.js/pull/15483#discussion_r1733224883.

TL;DR: Seb says ...
> https://forum.babylonjs.com/t/support-typescript-tsconfig-flag-exactoptionalpropertytypes-true/33824 will explain the details. We have since received a few PRs to fix it within various places in Babylon.

Reply:
From the PR that added the flag to Babylon:

> Indeed! They carry slightly different intent with `exactOptionalPropertyTypes: true`: a question mark allows it to be omitted but not set to undefined, while `| undefined` does the opposite. This can help catch a certain class of bugs (e.g., when checking stuff with `”foo” in object` incorrectly).